### DISCUSSION TITLE
How to deploy a serverless function in AWS lambda when using PNPM workspaces

### DISCUSSION THREAD

**Discussion by vibaiher-qatium** [pnpm/pnpm#6438] [Q&A] [answered]
Hello!

We have been using PNPM for a couple of months to build a monorepo. For now, we didn't included in our root workspace our lambda functions, but we were using already PNPM on them.

For deploying them, we were using this script:

```
pnpm run build # generates build folder, with traspilated code
pnpm install --frozen-lockfile --prod --node-linker=hoisted
cp --recursive --dereference node_modules build # copies and removes symlinks
rm -rf build/node_modules/.pnpm
```

After that, we zip the build folder and upload it to AWS. This wasn't an elegant solution, but it worked.

This script does not work now that we want to include those projects in the workspace, because the install command affects the root workspace node_modules folder. We tried several options, including pnpm deploy command, but we are not able to achieve to use it in combination with hoisted flag (AWS forces us to use a flat node_modules folder).

Is there something we are missing? Is it possible to include an option in deploy command to achieve that structure?

Thanks in advance!

**Comment by LuffyKing** [comment]
@vibaiher-qatium what solution did you end up with?

**Comment by KRE100** [comment, marked as answer]
pnpm --node-linker=hoisted install
https://pnpm.io/9.x/npmrc#node-linker
